warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PrintingVisitor.java

```diff
@@ -337,550 +345,1316 @@ private void childrenAccept(SimpleNode node, ParserVisitor visitor, Object data)
     }
 
     public Object visit(ASTJexlScript node, Object data) {
+        if (termsPrinted >= maxTermsToPrint) {
+            return null; // return early without printing for each additional node visited after the threshold
+        }
+        termsPrinted++;
+
+        if (termsPrinted >= maxTermsToPrint) {
+            LOGGER.trace("reached max terms for print threshold of " + maxTermsToPrint); // just log the threshold, logging the whole node for a very large
+                                                                                         // query could be less than constructive
+        }
```

Comment:
Wouldn't it be better to consolidate all the duplicate instances of this logic with a single method call?

Reply:
Ok, I updated the code to pull out the duplicates into a helper.